daemon/policy.go

```diff
@@ -220,17 +221,36 @@ func (d *Daemon) PolicyAdd(rules policyAPI.Rules, opts *AddOptions) (uint64, err
 	}
 
 	d.policy.Mutex.Lock()
+
 	// removedPrefixes tracks prefixes that we replace in the rules. It is used
 	// after we release the policy repository lock.
 	var removedPrefixes []*net.IPNet
+
+	// policySelectionWG is a waitgroup which is used to signal when the updating
+	// of all of the caches of endpoints in the rules which were added / updated
+	// have been updated.
+	var policySelectionWG sync.WaitGroup
+
+	// Get all endpoints at the time rules were added / updated so we can figure
+	// out which endpoints to regenerate / bump policy revision.
+	allEndpoints := endpointmanager.GetEndpoints()
+	epsIdentityConsumers := make([]policy.IdentityConsumer, len(allEndpoints))
+
+	// Need to explicitly convert endpoints to IdentityConsumers.
+	for _, ep := range allEndpoints {
```

Comment:
Update comment with reference to https://github.com/golang/go/wiki/InterfaceSlice